src/executor-group.jl

```diff
@@ -244,3 +410,5 @@ function get_grads(symbol, param_names, arg_names, data_names, inputs_need_grad,
 
   return grad_req_dict, freeze_idx
 end
+
+_merge_multi_context(outputs) = [concatenate(tensors, always_copy=false) for tensors in outputs]
```

Comment:
Agreed. It started as rather long python function, and simplified to one line. It should be refactored away now.